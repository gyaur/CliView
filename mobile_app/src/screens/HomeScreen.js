import React, { useState, useEffect } from 'react';
import { View, Button, TextInput, StyleSheet, Text, Image } from 'react-native';
import SearchBar from '../components/searchBar';
import Player from '../components/player';
import VolumeSlide from '../components/volumeSlide';
//import StaticServer from 'react-native-static-server';
import axios from 'axios'


const HomeScreen = () => {

  const [link, setLink] = useState('');
  const [IP, setIP] = useState('http://localhost:5000');
  const [IPInput, setIPInput] = useState(false)


  return (
    <View>
      <View>
        <Image
          style={{
            height: 150, marginHorizonta: 20, marginTop: 10,
          }}
          source={require('../assets/logo.png')}
        />
      </View>

      <SearchBar
        link={link}
        onLinkChange={newLink => setLink(newLink)}
        onLinkSubmit={() => {


          axios.post(IP + '/stream', { url: link })
            .then(function (response) {
              console.log(response);
            })
            .catch(function (error) {
              console.log(error);
            })
          console.log("link submitted : ", link)
        }

        }
      />

      <View style={styles.castButton}>
        <Button
          title="   Cast now   "
          color="#576574"
          style={{ margin: 30 }}
          onPress={() => {
            axios.post(IP + '/stream', { "url": link })
              .then(function (response) {
                console.log(response);
              })
              .catch(function (error) {
                console.log(error);
              })
          }
          }
        />
        <Text style={{ marginHorizontal: 15, color: '#c8d6e5' }}> Or</Text>
        <Button
          title="Add to queue"
          color="#576574"
          onPress={() => {
            //casting function here 
            axios.post(IP + '/queue ', { url: link })
              .then(function (response) {
                console.log(response);
              })
              .catch(function (error) {
                console.log(error);
              })
            console.log("link submitted : ", link)
          }}
        />
      </View >

      <View style={{
        flexDirection: 'row',
        marginTop: 35,
        justifyContent: "center",
        alignItems: "center",
      }}>

        <Button
          title="Skip to the next video"
          color="#576574"
          onPress={() => {  
            axios.post(IP + '/skip', {})
            .then(function (response) {
              console.log(response);
            })
            .catch(function (error) {
              console.log(error);
            })

          }} 
        />
      </View>


      <Player newIP={IP} />


      <View style={styles.slidecontainer}>
        <VolumeSlide newIP={IP} />
      </View>


      <View style={{
        marginLeft: "auto",
        marginRight: "auto",
        marginTop: 50,
        flexDirection: "row",
        backgroundColor: '#F0EEEE',
        height: 40,
        borderRadius: 7,
      }}>
        <Button
          title="Change IP"
          color="#576574"
          onPress={() => { IPInput ? setIPInput(false) : setIPInput(true) }}
        />

        <View>
          {IPInput ?
            <TextInput
              autoCapitalize="none"
              autoCorrect={false}
              placeholder="IP"
              style={styles.IPinputStyle}
              value={IP}
              onChangeText={newIP => setIP(newIP)}
            /> : null}

        </View>
      </View>

    </View>
  )
};

const styles = StyleSheet.create({

  castButton: {
    marginTop: 15,
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
  },
  slidecontainer: {
    marginTop: 30,
    display: "flex",
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
    width: "90%"
  },
  IPinputStyle: {
    flex: 1,
    fontSize: 14,
    marginHorizontal: 10
  }
});

export default HomeScreen;