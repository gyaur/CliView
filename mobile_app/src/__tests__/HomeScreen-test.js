import 'react-native';
import React from 'react';
import HomeScreen from '../screens/HomeScreen';
import renderer from 'react-test-renderer';

test('HomeScreen snapshot', ()=>{
    
    const snap = renderer.create(
        <HomeScreen/>
    ).toJSON();
   
    expect(snap).toMatchSnapshot();
});