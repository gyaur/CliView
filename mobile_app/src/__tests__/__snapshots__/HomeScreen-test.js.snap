// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "gray",
        "borderRadius": 7,
        "height": 100,
        "marginHorizontal": 15,
        "marginTop": 25,
      }
    }
  >
    <Text>
      Logo
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#F0EEEE",
        "borderRadius": 7,
        "flexDirection": "row",
        "height": 50,
        "marginHorizontal": 15,
        "marginTop": 15,
      }
    }
  >
    <Text />
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      onEndEditing={[Function]}
      placeholder="URL"
      rejectResponderTermination={true}
      style={
        Object {
          "flex": 1,
          "fontSize": 18,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 15,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#576574",
              },
            ]
          }
        >
          Cast now
        </Text>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#c8d6e5",
          "marginHorizontal": 10,
        }
      }
    >
       Or
    </Text>
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#576574",
              },
            ]
          }
        >
          Add to queue
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": 50,
      }
    }
  >
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 50,
            "flex": 1,
            "margin": 15,
            "marginLeft": 25,
            "opacity": 1,
            "paddingTop": 15,
          }
        }
      >
        <Text />
      </View>
    </View>
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 80,
            "flex": 1,
            "opacity": 1,
            "padding": 15,
          }
        }
      >
        <Text />
      </View>
    </View>
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 50,
            "flex": 1,
            "margin": 15,
            "marginRight": 25,
            "opacity": 1,
            "paddingTop": 15,
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 50,
        "width": "90%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "90%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexGrow": 1,
            "marginHorizontal": 10,
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <Text />
      </View>
      <RNCSlider
        disabled={false}
        enabled={true}
        inverted={false}
        maximumTrackTintColor="#576574"
        maximumValue={100}
        minimumTrackTintColor="#c8d6e5"
        minimumValue={0}
        onChange={[Function]}
        onRNCSliderSlidingComplete={[Function]}
        onRNCSliderSlidingStart={null}
        onRNCSliderValueChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        step={10}
        style={
          Array [
            Object {
              "height": 40,
            },
            Object {
              "flexGrow": 1,
              "height": 40,
              "width": 250,
            },
          ]
        }
        thumbImage={
          Object {
            "__packager_asset": true,
            "height": 1,
            "scale": 1,
            "uri": "uri",
            "width": 1,
          }
        }
        thumbTintColor="#c8d6e5"
        value={50}
      />
    </View>
  </View>
</View>
`;
